torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -795,12 +801,37 @@ struct PythonPrintImpl {
     }
   }
 
+  static bool containsNonASCIIString(const IValue& val) {
+    // TODO: This change breaks forward compatibility.
+    // Temporarily turning off serialization (torch.save only) of non ASCII
+    // string literals. We will turn it on a week.
+    return false;
+
+    bool hasNonASCII = false;
+    auto checkSubvalue = [&hasNonASCII](const IValue& val) {
+      if (val.isString()) {
+        const auto maxASCII = 0x7fu;
+        for (auto& c : val.toStringRef()) {
+          if (c > maxASCII) {
+            hasNonASCII = true;
+            return true;
+          }
+        }
+      }
+      return false;
+    };
+
+    val.visit(checkSubvalue);
+    return hasNonASCII;
+  }
+
   void printConstant(TaggedStringStream& stmt, const IValue& v) {
     const auto customFormatter = [&](std::ostream& ss, const IValue& v) {
-      if (v.isTensor()) {
-        ss << "CONSTANTS.c" << getOrAddTensorConstant(v.toTensor());
+      if (v.isTensor() || containsNonASCIIString(v)) {
```

Comment:
I feel like this isn't the right condition: we _should_ be able to correctly print non-ascii strings with the appropriate escapes. In fact, I think we actually do. I think what is going on here is that some aggregate constant is getting printed through a call to its pretty printer, and the bug lies in the fact that the pretty printer for that aggregate doesn't recursively invoke our our repr code for strings. I feel like this bug should be fixed there, this feels like a workaround.

Reply:
IMO,  only small string should be allowed to be inlined. All large constant  (lists,dicts, strings) should be in the constant table. I followed this logic and time is spent parsing string literals.  I would not spend time adding logic to parse unicode strings.